hoptimator-k8s/src/main/java/com/linkedin/hoptimator/k8s/K8sConfigProvider.java

```diff
@@ -0,0 +1,44 @@
+package com.linkedin.hoptimator.k8s;
+
+import java.io.StringReader;
+import java.util.Map;
+import java.util.Properties;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.kubernetes.client.openapi.models.V1ConfigMap;
+import io.kubernetes.client.openapi.models.V1ConfigMapList;
+
+import com.linkedin.hoptimator.ConfigProvider;
+
+
+public class K8sConfigProvider implements ConfigProvider {
+
+  private static final Logger log = LoggerFactory.getLogger(K8sConfigProvider.class);
+  public static final String HOPTIMATOR_CONFIG_MAP = "hoptimator-configmap";
+
+  // Loads top level configs and expands given properties under a file-like property key
+  // Ex:
+  //  log.properties: |
+  //    level=INFO
+  public Properties loadConfig(String... expansionFields) throws Exception {
+    Map<String, String> topLevelConfigs = loadTopLevelConfig(HOPTIMATOR_CONFIG_MAP);
+    Properties p = new Properties();
+    p.putAll(topLevelConfigs);
+    for (String expansionField : expansionFields) {
+      if (topLevelConfigs == null || !topLevelConfigs.containsKey(expansionField)) {
+        log.warn("Config map does not contain {}", expansionField);
+        continue;
+      }
+      p.load(new StringReader(topLevelConfigs.get(expansionField)));
```

Comment:
I guess this logic would repeat in any `ConfigProvider` impl. Probably better to have the SPI just return the raw properties, and then have `ConfigService` expand them this way. That way the SPI is easier to impl, and the expansion logic is in one place.

Reply:
Good point, though this does make throwing an exception instead trickier since there could be multiple config providers but we actually only intend to expand the properties coming from one them